mlir/lib/Conversion/TosaToRock/TosaToRock.cpp

```diff
@@ -632,6 +632,12 @@ struct TransposeRewritePattern : public OpRewritePattern<tosa::TransposeOp> {
     Value tInput = top.getOperand(0);
     Value tOutput = top.getResult();
 
+    // if the input has uses (apart from this one), we can't do this
+    if (!tInput.hasOneUse()) {
```

Comment:
Ah this needs to be moved inside output fusion cases... i.e where tInput is checked to be a conv or gemm.
The third one -- `mergeTransposeWithGemmLikeOp` -- still could run.